source/common/router/router.cc

```diff
@@ -125,6 +125,11 @@ FilterUtility::finalTimeout(const RouteEntry& route, Http::HeaderMap& request_he
   if (grpc_request && route.maxGrpcTimeout()) {
     const std::chrono::milliseconds max_grpc_timeout = route.maxGrpcTimeout().value();
     std::chrono::milliseconds grpc_timeout = Grpc::Common::getGrpcTimeout(request_headers);
+    if (route.grpcTimeoutOffset()) {
+      grpc_timeout =
+          std::max(std::chrono::milliseconds::zero(), (grpc_timeout - *route.grpcTimeoutOffset()));
```

Comment:
Presumably `std::chrono::milliseconds` is signed and this won't wrap around? Can you potentially add a comment? 

Also, if we end up setting this to zero, it will effectively disable the timeout, right? Should the default in this case actually be to use the supplied timeout in the header? That seems safer? I could go either way. Whatever you choose, can you update the docs to say what happens if the value is larger than the supplied header?

Reply:
Accoriding to https://en.cppreference.com/w/cpp/chrono/duration the value is signed:

```
std::chrono::milliseconds | duration</*signed integer type of at least 45 bits*/, std::milli>
```

There's code further down that handles the case where the timeout ends up at zero: it changes it to the value of `route.maxGrpcTimout`. That said, it does seem safer to not apply the offset in that case, because it would end up increasing the timeout instead of decreasing it which is likely the intended result